### PR TITLE
Couchbase - Get Destination Cluster Reference

### PR DESCRIPTION
**Pull request by mihaitusa**
Added new operation
Signed off by Tusa Mihai: tusa.mihai@gmail.com

### CODE REVIEW

**Comment by mihaitusa** [comment]
@tethryus ... do your magic...

File: cs-couchbase/src/main/java/io/cloudslang/content/couchbase/actions/xdcr/GetDestinationClusterReference.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
+/*
+ * (c) Copyright 2017 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+*/
+/*******************************************************************************
```
Comment (Bonczidai) [inline cs-couchbase/src/main/java/io/cloudslang/content/couchbase/actions/xdcr/GetDestinationClusterReference.java]: duplicate licence?

File: cs-couchbase/src/main/java/io/cloudslang/content/couchbase/factory/cluster/ClusterUriFactory.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import static io.cloudslang.content.couchbase.entities.couchbase.CouchbaseApi.POOLS_DEFAULT;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+
+public class ClusterUriFactory {
+    private ClusterUriFactory() {
+        // prevent instantiation
+    }
+
+    public static String getClusterUriValue(InputsWrapper wrapper) {
+        String action = wrapper.getCommonInputs().getAction();
```
Comment (Bonczidai) [inline cs-couchbase/src/main/java/io/cloudslang/content/couchbase/factory/cluster/ClusterUriFactory.java]: is this nullable? In that case maybe you should check

File: cs-couchbase/src/main/java/io/cloudslang/content/couchbase/utils/InputsUtil.java [hunk 3]
```diff
@@ -65,13 +69,13 @@ public static HttpClientInputs getHttpClientInputs(String username, String passw
...
-        httpClientInputs.setQueryParamsAreURLEncoded(FALSE);
+        httpClientInputs.setQueryParamsAreURLEncoded(valueOf(FALSE));
         httpClientInputs.setUsername(username);
         httpClientInputs.setPassword(password);
 
-        httpClientInputs.setTrustAllRoots(valueOf(getEnforcedBooleanCondition(trustAllRoots, true)));
-        httpClientInputs.setKeepAlive(valueOf(getEnforcedBooleanCondition(keepAlive, true)));
-        httpClientInputs.setUseCookies(valueOf(getEnforcedBooleanCondition(useCookies, true)));
+        httpClientInputs.setTrustAllRoots(valueOf(getEnforcedBooleanCondition(trustAllRoots, FALSE)));
+        httpClientInputs.setKeepAlive(valueOf(getEnforcedBooleanCondition(keepAlive, TRUE)));
```
Comment (Bonczidai) [inline cs-couchbase/src/main/java/io/cloudslang/content/couchbase/utils/InputsUtil.java]: do we need these tricks with objects? maybe we can just rely on boxing

Comment (mihaitusa) [reply cs-couchbase/src/main/java/io/cloudslang/content/couchbase/utils/InputsUtil.java]: You're right... I'll find a way to simplify it.

File: cs-couchbase/src/main/java/io/cloudslang/content/couchbase/actions/xdcr/GetDestinationClusterReference.java (hunk 1 above)
Comment (mihaitusa) [reply cs-couchbase/src/main/java/io/cloudslang/content/couchbase/actions/xdcr/GetDestinationClusterReference.java]: Well... maybe @tethryus can tell why...

File: cs-couchbase/src/main/java/io/cloudslang/content/couchbase/factory/cluster/ClusterUriFactory.java (hunk 2 above)
Comment (mihaitusa) [reply cs-couchbase/src/main/java/io/cloudslang/content/couchbase/factory/cluster/ClusterUriFactory.java]: wrapper object is not null... it contains at least http client minimal settings (defaults) and action string is set through @Action without user interference

**Pull request merged**